src/main/java/org/example/src/game/SaveSystem.java

```diff
@@ -0,0 +1,82 @@
+package org.example.src.game;
+
+import org.example.src.units.*;
+
+import java.io.File;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+public class SaveSystem {
+    protected static  File heroFile;
+    protected static Scanner heroFileScanner;
+
+    protected static  File unitFile;
+    protected static Scanner unitFileScanner;
+
+    public SaveSystem() throws IOException {
+        heroFile = new File("hero.csv");
+        unitFile = new File("unit.csv");
+        if (!heroFile.exists()) {
+            heroFile.createNewFile();
+        }
+        if (!unitFile.exists()) {
+            unitFile.createNewFile();
+        }
+
+        heroFileScanner = new Scanner(heroFile);
+        unitFileScanner = new Scanner(unitFile);
+    }
+
+    public void save(Unit unit ,Hero hero) throws IOException {
+        save(unit,unitFile);
+        save(hero,heroFile);
+    }
+    private void save(Unit object , File unitFile) throws IOException {
+        try {
+            FileWriter writer = new FileWriter(unitFile);
+            writer.write(object.toCSV());
+            writer.close();
+        } catch (IOException e) {
+            System.out.println("Error with save");
+            System.out.println(unitFile.getCanonicalPath());
+        }
+    }
+    public Unit loadUnit(){
+        List<String>unitArrayList = loadFile(unitFileScanner);
+        return switch (unitArrayList.getFirst()) {
+            case "Dragon" -> new Dragon(unitArrayList);
+            case "Goblin" -> new Goblin(unitArrayList);
+            case "Imp" -> new Imp(unitArrayList);
+            default -> null;
```

Comment:
Wouldn't this mean if you have a file that someone tampered with, in this case changed the name of the mob, then the save file wouldn't throw an exception and instead pass a null value to the opponent atribute?